packages/reactive_ble_mobile/ios/Classes/ReactiveBle/Tasks/CharacteristicNotify/CharacteristicNotifyTaskController.swift

```diff
@@ -4,22 +4,29 @@ struct CharacteristicNotifyTaskController: PeripheralTaskController {
 
     typealias TaskSpec = CharacteristicNotifyTaskSpec
 
-    private let task: Task
+    private let task: SubjectTask
 
-    init(_ task: Task) {
+    init(_ task: SubjectTask) {
         self.task = task
     }
 
-    func start(characteristic: CBCharacteristic) -> Task {
-        characteristic.service.peripheral.setNotifyValue(task.params.state.isOn, for: characteristic)
+    func start(characteristic: CBCharacteristic) -> SubjectTask {
+        guard let peripheral = characteristic.service?.peripheral
+        else { return task.with(state: task.state.finished(CharacteristicNotifyError.unExpected)) }
+        
+        peripheral.setNotifyValue(task.params.state.isOn, for: characteristic)
         return task.with(state: task.state.processing(.applying))
     }
 
-    func cancel(error: Error) -> Task {
+    func cancel(error: Error) -> SubjectTask {
         return task.with(state: task.state.finished(error))
     }
 
-    func complete(error: Error?) -> Task {
+    func complete(error: Error?) -> SubjectTask {
         return task.with(state: task.state.finished(error))
     }
+
+    private enum CharacteristicNotifyError: Error{
+            case unExpected;
+            }
```

Comment:
Holly cow, what's with formatting here? Okay, it must be alcohol in my blood 🙈

Reply:
yes I did even change this to a "normal" formatting but Xcode automatically changes it back to this crap 👀